invenio_app_ils/documents/jsonresolvers/document_circulation.py

```diff
@@ -82,8 +82,20 @@ def circulation_resolver(document_pid):
             if bucket["key"] in "FOR_REFERENCE_ONLY":
                 items_for_reference_count = bucket["doc_count"]
 
+        active_ill_loans_count = 0
+
+        for hit in loans_result.hits:
```

Comment:
you don't need to count the loans in the `for` loop. You could do something like (pseudocode):
```python

active_ill_loans_count = loans_search.filter(item_pid__type=BORROWING_REQUEST_PID_TYPE).count()
```

Reply:
Actually it was myself who suggested to get rid of an extra search, since it adds more delay to circulation. What do you think?

Reply:
discussed IRL, agreed that this solution is faster